src/tcp_axtls.c

```diff
@@ -387,20 +401,31 @@ int tcp_ssl_read(struct tcp_pcb *tcp, struct pbuf *p) {
     } else if(read_bytes > 0){
       if(fd_data->on_data){
         fd_data->on_data(fd_data->arg, tcp, read_buf, read_bytes);
+        // fd_data may have been freed in callback
+        fd_data = tcp_ssl_get(tcp);
+        if(NULL == fd_data)
+          return SSL_CLOSE_NOTIFY;
```

Comment:
Can you comment on where the actual fix is? I see a lot of changed lines, but I'd like to know where the fixed happened so that I can review this accurately.

Reply:
The fix is at line 418 through 420 and 404 through 407.
The problem occurs when the client does a delete on _client from a callback and fd_data becomes invalid.